Szwalnia/NowyRegal.cs

```diff
@@ -44,20 +43,26 @@ private void btnDodajRegal_Click(object sender, EventArgs e)
                 regalNew.Oznaczenie = txtOznaczenie.Text;
                 db.Regaly.Add(regalNew);
                 db.SaveChanges();
-                MessageBox.Show("Pomyślnie dodano do bazy regał " + nowy);
-
-                Regaly_wykaz wykaz = new Regaly_wykaz(); //to na razie jest tylko w celu sprawdzenia czy się dodał i pokazuje alfabetycznie
-                wykaz.Show();
+                MessageBox.Show("Pomyślnie dodano do bazy regał " + txtOznaczenie.Text);
                 this.Close();
             }
             else
-                MessageBox.Show("Regał " + nowy + " już istnieje!");
+                MessageBox.Show("Regał " + txtOznaczenie.Text + " już istnieje!");
                 regalNew = null;    
         }
 
         private void NowyRegal_FormClosed(object sender, FormClosedEventArgs e)
         {
-            Start.GetForm.Show();
+            if (Application.OpenForms["NowaPolka"] == null)
```

Comment:
jeśli już chcemy używać kolekcji `OpenForms` to zamiast stringa proponuję `typeof(NowaPolka).Name`

natomiast zanim to dopuszczę proszę wytłumaczyć co chcemy tutaj osiągnąć? jaki jest algorytm działania?

Reply:
Chodzi o to, że NowyRegal może być otwarty albo bezpośrednio z UkladMagazynu albo z NowaPolka. Chciałabym, żeby po zamknięciu aplikacja wracała na odpowiedni formularz